Creating port from `uint16` field fails

@ynadji reported the following issue over Slack:


```ruby
module foo;

import spicy;

public type X = unit {
    p: uint16;
    on %done {
        print f(self);
    }
};

public function f(x: X): tuple<p: port> {
    # BUG: creating port fails.
    local p = port(x.p, spicy::Protocol::TCP);
    return (p,);
}
```

This fails during compilation with an error like this:

```console
/private/var/folders/ht/vtb0rkbd4ws15vx02xdb84zw0000gn/T/foo_1451c101ef254fb1-f741d8d37d7e852a.cc:215:48: error: no matching function for call to 'valueOrInit'
    ::hilti::rt::Port p = ({ ::hilti::rt::Port(::hilti::rt::optional::valueOrInit((*x).p), ::hilti::rt::Protocol::TCP); }
                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/Cellar/spicy/HEAD-ba586ba/include/hilti/rt/types/optional.h:49:14: note: candidate template ignored: deduced type 'optional<...>' of 1st parameter does not match adjusted type 'const optional<...>' of argument [with T = SafeInt<unsigned
short, hilti::rt::integer::detail::SafeIntException>]
inline auto& valueOrInit(std::optional<T>& t) {
             ^
/usr/local/Cellar/spicy/HEAD-ba586ba/include/hilti/rt/types/optional.h:41:14: note: candidate function template not viable: requires 2 arguments, but 1 was provided
inline auto& valueOrInit(std::optional<T>& t, const T& default_) {
             ^
/private/var/folders/ht/vtb0rkbd4ws15vx02xdb84zw0000gn/T/foo_1451c101ef254fb1-f741d8d37d7e852a.cc:215:23: error: no viable conversion from 'void' to '::hilti::rt::Port'
    ::hilti::rt::Port p = ({ ::hilti::rt::Port(::hilti::rt::optional::valueOrInit((*x).p), ::hilti::rt::Protocol::TCP); }
                      ^   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/Cellar/spicy/HEAD-ba586ba/include/hilti/rt/types/port.h:42:5: note: candidate constructor not viable: cannot convert argument of incomplete type 'void' to 'const hilti::rt::Port &' for 1st argument
    Port(const Port&) = default;
    ^
/usr/local/Cellar/spicy/HEAD-ba586ba/include/hilti/rt/types/port.h:43:5: note: candidate constructor not viable: cannot convert argument of incomplete type 'void' to 'hilti::rt::Port &&' for 1st argument
    Port(Port&&) noexcept = default;
    ^
2 errors generated.
[error] spicyc: JIT compilation failed
```

If one instead creates a local `uint16` variable in lieu of accessing the unit field directly this works.

```ruby
public function f(x: X): tuple<p: port> {
    # WORKAROUND: creating port via local uint16 works.
    local pp: uint16 = x.p;
    local p = port(pp, spicy::Protocol::TCP);
    return (p,);
}
```

--- a/hilti/toolchain/src/compiler/codegen/expressions.cc
+++ b/hilti/toolchain/src/compiler/codegen/expressions.cc
@@ -41,7 +41,7 @@ struct Visitor : hilti::visitor::PreOrder<std::string, Visitor> {
         cxx::Block block;
         cg->pushCxxBlock(&block);
         auto arguments =
-            util::join(node::transform(n.arguments(), [this](auto& x) { return cg->compile(x, true); }), ", ");
+            util::join(node::transform(n.arguments(), [this](auto& x) { return cg->compile(x, lhs); }), ", ");
         cg->popCxxBlock();
 
         block.addStatement(fmt("%s(%s)", cxx::ID(n.cxxname()), arguments));

